### PR TITLE
S02_US08 Createaccount part1

### PR DESCRIPTION
**Pull request by Wowlfy**

### CODE REVIEW

**Review by maxime-guichard — APPROVED** [review]

File: src/Security/UserfixturesAuthenticator.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+use Symfony\Component\Security\Core\Security;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Csrf\CsrfToken;
+use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
+use Symfony\Component\Security\Guard\Authenticator\AbstractFormLoginAuthenticator;
+use Symfony\Component\Security\Guard\PasswordAuthenticatedInterface;
+use Symfony\Component\Security\Http\Util\TargetPathTrait;
+
+class UserfixturesAuthenticator extends AbstractFormLoginAuthenticator implements PasswordAuthenticatedInterface
```
Comment (LucyMerlier) [inline src/Security/UserfixturesAuthenticator.php:23]: why does your authenticator have the word "fixtures" in its name?

File: templates/security/login.html.twig [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{% extends 'base.html.twig' %}
```
Comment (LucyMerlier) [inline templates/security/login.html.twig:1]: why does this extend base.html.twig directly?

File: templates/security/login.html.twig [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    {#
+        Uncomment this section and add a remember_me option below your firewall to activate remember me functionality.
+        See https://symfony.com/doc/current/security/remember_me.html
+
+        <div class="checkbox mb-3">
+            <label>
+                <input type="checkbox" name="_remember_me"> Remember me
+            </label>
+        </div>
+    #}
```
Comment (LucyMerlier) [inline templates/security/login.html.twig:36]: comments are good, but commented code is bad => leave no commented code in your files, even when it's auto-generated by Symfony

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
see comments

**Review by maxime-guichard — APPROVED** [review]

File: src/DataFixtures/UserFixtures.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        // Création d’un utilisateur de type “compagny”
+        $compagny = new User();
+        $compagny->setEmail('compagny@monsite.com');
+        $compagny->setRoles(['ROLE_COMPAGNY']);
+        $compagny->setPassword($this->passwordEncoder->encodePassword(
+            $compagny,
+            'compagnypassword'
+        ));
+
+        $manager->persist($compagny);
```
Comment (LucyMerlier) [inline src/DataFixtures/UserFixtures.php:46]: in english : "company" => no "g"

File: templates/security/login.html.twig [hunk 5]
```diff
@@ -0,0 +1,30 @@
+{% extends 'layout.html.twig' %}
+
+{% block title %}Log in!{% endblock %}
```
Comment (LucyMerlier) [inline templates/security/login.html.twig:3]: so, you website is gonna be in english after all?

File: templates/security/login.html.twig [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+{% block body %}
+<form method="post">
+    {% if error %}
+        <div class="alert alert-danger">{{ error.messageKey|trans(error.messageData, 'security') }}</div>
+    {% endif %}
+
+    {% if app.user %}
+        <div class="mb-3">
+            You are logged in as {{ app.user.username }}, <a href="{{ path('app_logout') }}">Logout</a>
```
Comment (LucyMerlier) [inline templates/security/login.html.twig:13]: same as above

File: templates/security/login.html.twig [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        </div>
+    {% endif %}
+
+    <h1 class="h3 mb-3 font-weight-normal">Please sign in</h1>
```
Comment (LucyMerlier) [inline templates/security/login.html.twig:17]: same as above

File: templates/security/login.html.twig [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    <label for="inputEmail">Email</label>
+    <input type="email" value="{{ last_username }}" name="email" id="inputEmail" class="form-control" autocomplete="email" required autofocus>
+    <label for="inputPassword">Password</label>
```
Comment (LucyMerlier) [inline templates/security/login.html.twig:20]: same as above

File: templates/security/login.html.twig [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    <input type="password" name="password" id="inputPassword" class="form-control" autocomplete="current-password" required>
+
+    <input type="hidden" name="_csrf_token"
+           value="{{ csrf_token('authenticate') }}"
+    >
+    <button class="btn btn-lg btn-primary" type="submit">
+        Sign in
```
Comment (LucyMerlier) [inline templates/security/login.html.twig:27]: same as above

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
see comments

**Review by maxime-guichard — APPROVED** [review]

File: templates/security/login.html.twig [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+{% block body %}
+<form method="post">
+    {% if error %}
+        <div class="alert alert-danger">{{ error.messageKey|trans(error.messageData, 'security') }}</div>
+    {% endif %}
+
+    {% if app.user %}
+        <div class="mb-3">
+            Tu es connecté en tant que  {{ app.user.username }}, <a href="{{ path('app_logout') }}">Deconnexion</a>
```
Comment (LucyMerlier) [inline templates/security/login.html.twig:13]: "Vous" may be better here + be inclusive ! ("connecté·e")

**Review by LucyMerlier — CHANGES_REQUESTED** [review]

**Review by Erdnaxela45 — APPROVED** [review]

**Review by LucyMerlier — APPROVED** [review]

**Pull request merged**